cgsmiles/graph_layout.py

```diff
@@ -1,347 +1,176 @@
-        avg_dist = avg_dist / len(self.graph.edges)
-
-        for node in pos:
-            pos[node] *= self.default_bond / avg_dist
-        return pos
-
-    def circular(self):
-        positions = generate_circle_coordinates(radius=self.default_bond*2,
-                                                num_points=len(self.graph))
-
-        pos = {}
-        start = list(self.graph.nodes)[0]
-        for idx, (node, next_node) in enumerate(nx.find_cycle(self.graph, source=start)):
-            pos[node] = positions[idx]
-        return pos
-
-    def rotate_to_diagonal(self, positions):
-        # get all the distances
-        distances = squareform(pdist(positions, 'euclidean'))
-        # find pair belonging to largest distance
-        pair = unravel_index(distances.argmax(), distances.shape)
-        # get vector
-        vector = positions[pair[0]] - positions[pair[1]]
-        # compute angle
-        if self.align_with == 'diag':
-            diagonal = np.array(self.bounding_box)
-        elif self.align_with == 'x':
-            diagonal = np.array([1,0])
-        elif self.align_with == 'y':
-            diagonal = np.array([0, 1])
-
-        angle = np.arctan2(diagonal[1], diagonal[0]) - np.arctan2(vector[1], vector[0])
-        # rotate object onto diagonal
-        rotation_matrix = np.array([[np.cos(angle), -np.sin(angle)],
-                                    [np.sin(angle), np.cos(angle)]])
-
-        # Apply the rotation matrix to all positions
-        rotated_positions = np.dot(positions, rotation_matrix.T)
-        return rotated_positions
-
-    def md_layout(self):
-        counter = 0
-        while counter < self.max_iter:
-            if self.circle:
-                pos = self.circular()
-            else:
-                pos = self.vsepr_layout()
-            pos, energy = self.force_minimize(pos)
-            if energy < self.target_energy:
-                break
-            counter += 1
-
-        # rotate molecule to fit into bounding box
-        if self.bounding_box:
-            pos_aligned = self.rotate_to_diagonal(pos)
-
-        # write positions as dict of graph node keys
-        positions = {}
-        for node_key, idx in self.atom_to_idx.items():
-            positions[node_key] = pos_aligned[idx]
-        return positions
+    dict
+        a dict of positions
+    """
+    atom_to_idx = OrderedDict(zip(list(graph.nodes), range(0, len(graph.nodes))))
+    max_iter = 5
+    counter = 0
+    while counter < max_iter:
+        pos = vespr_layout(graph, default_bond)
+        pos, energy = _force_minimize(graph,
+                                      pos,
+                                      default_bond,
+                                      default_angle,
+                                      atom_to_idx,
+                                      lbfgs_options)
+        if energy < target_energy:
+            break
+        counter += 1
```

Comment:
when you do a lot of consecutive drawings it is nice having the algorithm try a couple of times. Otherwise, you need to isolate they failed drawing and rerun it. It does not return the lowest energy option because all plots with energy > target_energy are usually crap.